src/Ether.Api/Types/ADPasswordValidator.cs

```diff
@@ -30,15 +30,15 @@ public Task ValidateAsync(ResourceOwnerPasswordValidationContext context)
             }
 
             var domain = domainAndName[0];
-            var userName = context.UserName.Substring(context.UserName.IndexOf('\\'));
+            var userName = domainAndName[1];
             if (type == ContextType.Machine)
             {
                 domain = Environment.MachineName;
             }
 
             using (var principal = new PrincipalContext(type, domain))
             {
-                var isValid = principal.ValidateCredentials(userName, context.Password);
+                var isValid = principal.ValidateCredentials(userName, context.Password, ContextOptions.Negotiate);
```

Comment:
It works without being in domain. but next code causes to error. My PS in in domain but it has also Administrator user that is not from domain. When i am login as .\Administrator, it passes validation in 
`principal.ValidateCredentials(userName, context.Password, ContextOptions.Negotiate)`
but than fails on next lines where code tryes to get user name from Environment, instead of user name that I have provided:
`var userNameToSearch = type == ContextType.Machine ? Environment.UserName : context.UserName;`
Because I am trying to login using local user different from current Windows user.

Reply:
@BerserkerDotNet Issue fixed by next commit. Verified using AD account and local machine account (./Administrator)